src/main/java/aeneas/views/BuildLevelView.java

```diff
@@ -71,27 +87,33 @@
 
   @Override
   public void initialize(URL location, ResourceBundle resources) {
-    bullpenView = new BullpenView(bullpenBox, this);
-
-    Piece testPiece = new Piece(new Square[] {
-        new Square(0, 0),
-        new Square(1, 0),
-        new Square(1, 1),
-        new Square(1, 2),
-        new Square(1, 3),
-        new Square(1, 4),
-    });
-
+    this.bullpenView = new BullpenView(bullpenBox, (Pane) this);
 
-    bullpenView.addPiece(testPiece, model);
-
-    boardView = new BoardView(levelModel.getBoard());
+    this.boardView = new BoardView(levelModel.getBoard());
     VBox.setMargin(boardView, new Insets(10, 10, 10, 10));
     centerBox.setAlignment(Pos.TOP_RIGHT);
     centerBox.getChildren().add(boardView);
 
     saveButton.setOnMouseClicked(
         new SaveLevelController(mainView, levelModel));
 
+    piecePickerDialog.setTransitionType(DialogTransition.CENTER);
+
+    addPiece.setOnMouseClicked((e) -> {
+      piecePickerDialog.show(this);
+      piecesPane.getChildren().clear();
+
+      for (Piece pieceModel : PieceFactory.pieces) {
+        PieceView pView = new PieceView((Pane) this, pieceModel, model, PIECE_PICKER_SQUARE_SIZE);
+        piecesPane.getChildren().add(pView);
+
+        pView.setOnMouseClicked((click) -> {
+          IMove move = new AddPieceMove(levelModel.getBullpen(), pieceModel.clone());
+          if (move.execute()){
```

Comment:
Should also be adding `move` to the undo stack (`Model.addNewMove()` I believe--I'm not sure if it is actually implemented yet, but the method should be there).